src/main/java/com/amplitude/HttpCall.java

```diff
@@ -79,6 +82,8 @@ protected Response syncHttpCallWithEventsBuffer(List<Event> events) {
       }
       JSONObject responseJson = new JSONObject(sb.toString());
       responseBody = Response.populateResponse(responseJson);
+    } catch(AmplitudeInvalidAPIKeyException e) {
```

Comment:
if we are catching the exception already, this method is not throwing any more exception I guess

Reply:
Actually, the exception will throw in here. And the catch in Amplitude.java is not doing anything but required surrounded the try-catch block from ItelliJ.

Reply:
the exception is thrown in `Response.populateResponse` right? And as it is get caught here, this method is not throwing any more `AmplitudeInvalidAPIKeyException`, the Intellij only gives hint since `throws AmplitudeInvalidAPIKeyException` is in signature. So if you want to throw again at here, you may want to add like `throw e` right? I guess if we want to support hard stop, then we don't need to log for this.